aten/src/ATen/native/native_functions.yaml

```diff
@@ -170,18 +170,30 @@
     CPU: _assert_async_cpu
     CUDA: _assert_async_cuda
 
+- func: _functional_assert_async(Tensor self, Tensor dep_token) -> Tensor
+  dispatch:
+    CPU: _functional_assert_async_cpu
+
 - func: _assert_async.msg(Tensor self, str assert_msg) -> ()
   dispatch:
     CPU: _assert_async_msg_cpu
     CUDA: _assert_async_msg_cuda
 
+- func: _functional_assert_async.msg(Tensor self, str assert_msg, Tensor dep_token) -> Tensor
+  dispatch:
+    CPU: _functional_assert_async_msg_cpu
+
 - func: _assert_tensor_metadata(Tensor a, SymInt[]? size=None, SymInt[]? stride=None, ScalarType? dtype=None) -> ()
 
 - func: sym_constrain_range(Scalar size, int? min=None, int? max=None) -> ()
   dispatch:
     CPU: sym_constrain_range_cpu
     CUDA: sym_constrain_range_cuda
 
+- func: functional_sym_constrain_range(Scalar size, int? min=None, int? max=None, Tensor? dep_token=None) -> Tensor
```

Comment:
Why does it have take optional dep_token?

Reply:
Yeah this is a little bit annoying but since both `min` and `max` before it are both optional so I make it optional as well. The other option is to just move it before `min` / `max` arg but I feel like it makes it a little bit inconsistency with other functional assertions since `dep_token` is the last in others.
Not idea but let me add a runtime check for it to make sure `dep_token` is passed in?

Reply:
Haha I see what you mean. Hmm I think it is fine to have it before the min/max. I feel like we shouldn't make the dep_token implementation more complicated.